### PR TITLE
fix: Post submit error

### PR DESCRIPTION
**Pull request by singh-inder**

### CODE REVIEW

File: api/posts.js [hunk 1]
```diff
@@ -25,14 +25,7 @@ router.post("/", authMiddleware, async(req, res)=>{
 
         const post = await new PostModel(newPost).save();
        
-        const postCreated = await PostModel.findById(post._id).populate("user") //so thhat we can take them and render onto front end
-        return res.json(postCreated);   //since it is populated, its easy to connect to frontend by loking at its structure
```
Comment (singh-inder) [inline api/posts.js:29]: Select only one way to send back post. Here I'm sending back the post without populating any data.

File: api/posts.js [hunk 2]
```diff
@@ -25,14 +25,7 @@ router.post("/", authMiddleware, async(req, res)=>{
...
         const post = await new PostModel(newPost).save();
        
-        const postCreated = await PostModel.findById(post._id).populate("user") //so thhat we can take them and render onto front end
-        return res.json(postCreated);   //since it is populated, its easy to connect to frontend by loking at its structure
-        
-       // console.log(post);
-       //instead of sending any msg or whole post-> we will just send postId back to frontend(from backend). so that we can use it to display the post on viewport(layout)
-       //return res.json(post) or res.send ("Post created")
-       return res.json(post._id);
-
```
Comment (singh-inder) [inline api/posts.js:35]: This return statement is not reachable because you're already returning above.

File: utils/postActions.js [hunk 3]
```diff
@@ -10,15 +10,16 @@ const Axios = axios.create({
 });
 
 
-export async function submitNewPost(text, location, picUrl, setPosts, setNewPost, setError){
+export async function submitNewPost(user,text, location, picUrl, setPosts, setNewPost, setError){
```
Comment (singh-inder) [inline utils/postActions.js:13]: pass in user object in function parameters and use it to populate user data in new post.

**Comment by manjunadhkonapalli** [comment]
Solving undefined error of length

**Pull request merged**